mla/analysis.py

```diff
@@ -147,11 +145,19 @@ def produce_trial(analysis: Analysis, flux_norm: float = 0,
         np.random.seed(random_seed)
 
     background = analysis.model.inject_background_events()
+    background['time'] = analysis.model.scramble_times(
+        background['time'],
+        analysis.model.background_time_profile,
+    )
 
-    if flux_norm > 0 or n_signal_observed > 0:
+    if flux_norm > 0 or n_signal_observed is not None:
         signal = analysis.model.inject_signal_events(analysis.source,
                                                      flux_norm,
                                                      n_signal_observed)
+        signal['time'] = analysis.model.scramble_times(
```

Comment:
Why would injected signals require grl to weight time distribution?

Reply:
you're right. I should have a grl_weight flag or something to turn this on for background only

Reply:
I think we would still need to remove signal events that are outside grl after assign time based on the signal time distribution(and an option to disable the filter)